### PR TITLE
Update README.md

### PR DESCRIPTION
**Pull request by YanaXu**
The original `README.md` is not well organized. 
This PR is to update it:

- Add a section to list all parameters
- Group all workflow samples into one section
- Add steps to prepare service principal and managed identities before using them for login
- Add examples for login with managed identities

This PR will not be merged till the next release because the workflow of login with managed identities will only work in the new release.

### CODE REVIEW

**Comment by YanaXu** [comment]
Hi @MS-jeremy , @MoChilia , @jiasli , @isra-fel , @msJinLei , please help to review the update. Thanks.

**Comment by MS-jeremy** [comment]
> Hi @MS-jeremy , @MoChilia , @jiasli , @isra-fel , @msJinLei , please help to review the update. Thanks.

@YanaXu I'm not quite sure why I'm not the reviewer of this article, can you check if you @ the wrong person?

**Comment by YanaXu** [comment]
> > Hi @MS-jeremy
> >             
> >             Jeremy Li
> >             FTE , @MoChilia
> >             
> >             Shiying Chen
> >             FTE , @jiasli
> >             
> >             Jiashuo Li
> >             FTE , @isra-fel
> >             
> >             Yeming Liu
> >             FTE , @msJinLei
> >             
> >             Lei Jin
> >             FTE , please help to review the update. Thanks.
> 
> @YanaXu Yan Xu FTE I'm not quite sure why I'm the reviewer of this article, can you check if you @ the wrong person?

@MS-jeremy, sorry for this. I mistook you for the other Jeremy Li.

File: README.md [hunk 1]
```diff
@@ -102,154 +157,331 @@ jobs:
...
-Refer to the [Azure Stack Hub Login Action Tutorial](https://learn.microsoft.com/azure-stack/user/ci-cd-github-action-login-cli?view=azs-2008) for more detailed instructions.
+### Login With System-assigned Managed Identity
 
-## Configure deployment credentials
-  
-### Configure a service principal with a secret
+> [!NOTE]
+>
+> "Login With System-assigned Managed Identity" is only supported on GitHub self-hosted runners and the self-hosted runners need to be hosted by Azure virtual machines.
+
```
Comment (MoChilia) [inline README.md:313]:
```suggestion
> [!WARNING]
> Avoid using managed identity login on self-hosted runners in public repositories. Managed identities enable secure authentication with Azure resources and obtain Azure AD tokens without the need for explicit credential management. Any user can open pull requests against your repository and access your self-hosted runners without credentials. See more details in [self-hosted runner security](https://docs.github.com/en/actions/hosting-your-own-runners/managing-self-hosted-runners/about-self-hosted-runners#self-hosted-runner-security).
```
@jiasli, @Jacekey23, shall we add a warning message for using manage identity on self-hosted runners in public repositories?

File: README.md [hunk 2]
```diff
@@ -284,11 +535,29 @@ This action doesn't implement ```az logout``` by default at the end of execution
...
       az account clear
+
+- name: Azure PowerShell script
+  uses: azure/powershell@v1.2.0
+  with:
+    azPSVersion: "latest"
+    inlineScript: |
+      Clear-AzContext -Scope Process
+      Clear-AzContext -Scope CurrentUser
 ```
```
Comment (MoChilia) [inline README.md:547]:
```suggestion

## Azure CLI output
To suppress the output of Azure CLI commands after logging in, you can set the environment variable `AZURE_CORE_OUTPUT` to `none` at the Workflow, Job or Step level.

```yaml
- name: Azure CLI script
  uses: azure/CLI@v1
  env:
    AZURE_CORE_OUTPUT: none
  with:
    azcliversion: latest
    inlineScript: |
      az account show
      
```

@jiasli, @Jacekey23, shall we add `AZURE_CORE_OUTPUT: none` in each example or add a new section to guide users?

File: README.md [hunk 3]
```diff
@@ -1,87 +1,142 @@
...
+{
+    "clientSecret":  "******",
+    "subscriptionId":  "******",
+    "tenantId":  "******",
+    "clientId":  "******"
+}
 ```
 
-## Sample workflow that uses Azure login action using OIDC to run az cli (Linux)
+It's used in login with a Azure service principal secret.
```
Comment (isra-fel) [inline README.md:70]:
```suggestion
It's used in login with an Azure service principal.
```

File: README.md [hunk 4]
```diff
@@ -1,87 +1,142 @@
 # GitHub Actions for deploying to Azure
 
-## Automate your GitHub workflows using Azure Actions
-
 [GitHub Actions](https://help.github.com/articles/about-github-actions) gives you the flexibility to build an automated software development lifecycle workflow.
```
Comment (isra-fel) [inline README.md:3]: This doc should focus on the login action. Background information and other actions are less important.

**Review by isra-fel — CHANGES_REQUESTED** [review]
Very rich information here. A few suggestions:

1. A table of content (toc) would be useful to navigate through such a long doc
2. We should be clear about our recommended way of using the action - together with azure cli / ps action. Put it at the beginning.
3. Would be clearer to organize the parameters in a table

File: README.md (hunk 4 above)
Comment (YanaXu) [reply README.md]: Updated. Update the title and move GitHub Action and other actions' description to the reference section.

File: README.md (hunk 1 above)
Comment (MoChilia) [reply README.md:309]: cc @dcaro @yonzhan

File: README.md (hunk 2 above)
Comment (MoChilia) [reply README.md:543]: cc @dcaro @yonzhan

**Comment by YanaXu** [comment]
> 1. ble of content (toc) would be useful to navigate through such a long

Updated.

File: README.md [hunk 5]
```diff
@@ -1,87 +1,178 @@
...
+  - [Az logout and security hardening](#az-logout-and-security-hardening)
+  - [Azure CLI dependency](#azure-cli-dependency)
+  - [Reference](#reference)
+    - [GitHub Action](#github-action)
+    - [GitHub Actions for deploying to Azure](#github-actions-for-deploying-to-azure)
+    - [Azure CLI Action](#azure-cli-action)
+    - [Azure PowerShell Action](#azure-powershell-action)
+  - [Contributing](#contributing)
+
+With the [Azure Login Action](https://github.com/Azure/login), you can login to Azure and run Azure CLI and Azure PowerShell scripts.
```
Comment (isra-fel) [inline README.md:31]:
```suggestion
With the [Azure Login Action](https://github.com/Azure/login), you can login to Azure and run [Azure CLI](link to cli action) and [Azure PowerShell](link to ps action) scripts.
```

**Review by isra-fel — DISMISSED** [review]

File: README.md (hunk 1 above)
Comment (dcaro) [reply README.md:341]: We should have this message at the beginning of this article.

Comment (YanaXu) [reply README.md:341]: @dcaro, @MoChilia  Is the warning message content OK now?

Comment (MoChilia) [reply README.md:341]: It's good for me. Agree to have this message at the beginning.

File: README.md (hunk 5 above)
Comment (YanaXu) [reply README.md]: Updated.

**Comment by YanaXu** [comment]
The Comments are submerged. Here are the links. Other comments are resolved.

- [Yeming's comment](https://github.com/Azure/login/pull/363#pullrequestreview-1716900137)
  - Updated.
- Shiying's comments
  - [about MI warning ](https://github.com/Azure/login/pull/363#discussion_r1384295070) 
    - Added.
  - [about output warning](https://github.com/Azure/login/pull/363#discussion_r1384333383)

**Review by MoChilia — APPROVED** [review]

**Pull request merged**